spectrum/input.py

```diff
@@ -108,37 +108,53 @@ def create_blog_article(self, title, text='Lorem ipsum', image=None):
         # TODO: return id and/or node id
 
     def create_article_fragment(self, id, image):
-        create_url = "%s/admin/structure/article_fragment/add" % self._host
-        create_page = self._browser.get(create_url)
-        form = mechanicalsoup.Form(create_page.soup.form)
-        form.input({'name[0][value]': id})
-        form.attach({'files[image_0]': image})
+        filtered_content_url = "%s/admin/content?status=All&type=article&title=%s" % (self._host, id)
+        filtered_content_page = self._browser.get(filtered_content_url)
+
+        try:
+            view_url = "%s%s" % (self._host, filtered_content_page.soup.find('td', 'views-field-title').find('a', href=True, text=id).get('href'))
+            edit_url = "%s%s" % (self._host, filtered_content_page.soup.find('td', 'views-field-operations').find('li', 'edit').find('a', href=True, text='Edit').get('href'))
+        except (AttributeError, TypeError):
+            raise AssertionError('View and edit link not found for article: %s' % id)
+
         LOGGER.info(
-            "Submitting thumbnail %s",
-            image,
+            "Access edit form",
             extra={'id': id}
         )
-        self._choose_submit(form, 'image_0_upload_button', value='Upload')
-        response = self._browser.submit(form, create_page.url)
-        form = mechanicalsoup.Form(response.soup.form)
 
+        edit_page = self._browser.get(edit_url)
+
+        form = mechanicalsoup.Form(edit_page.soup.form)
 
-        form.attach({'files[banner_image_0]': image})
+        if edit_page.soup.find('input', {'name': 'field_image_0_remove_button'}):
+            self._choose_submit(form, 'field_image_0_remove_button', value='Remove')
+            response = self._browser.submit(form, edit_page.url)
+            form = mechanicalsoup.Form(response.soup.form)
+            LOGGER.info(
```

Comment:
usually I try to keep logs before the action they are taking, so that they log the intent before something explodes; this could go right before the related `self._browser.submit()`